analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -408,14 +386,15 @@ End Sub
         [TestMethod]
         public void CreateCfg_UndefinedSymbol_ReturnsCfg_CS()
         {
-            const string code = @"
-public class Sample
-{
-    public void Main()
-    {
-        Undefined(() => 45);
-    }
-}";
+            const string code = """
+                public class Sample
+                {
+                    public void Main()
+                    {
+                        Undefined(() => 45);
+                    }
+                }
+                """;
```

Comment:
Why not also `VerifyCfgCS` in this method? Due to the `CompileIgnoreErrorsCS`?
You could use `Compile` if you'd put `// Error CSxxx: Message` on the relevant line.

Reply:
This might be bit too hard if there's more than one error, like in the `DataRow` around. So you should consider doing this only for snippets with single error on a line.

Reply:
Yes and sure, I'll replace where there's only one compile error.

Reply:
In the `Compile` methods of the  `TestHelper` the options are to set the  `ignoreErrors` to true or false.
If I set it to false the compilation does not fail. If I leave it to true, the compilation fails even if I add the comment `// Error CSXXX I'm doing this on purpose`.
Being curious that in our UTs we just set the error comment and it works well -without specifying to ignore the compilation error - I found that we use the `SnipperCompiler`[ here](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompiler.cs#L53) that is less elastic.